### PR TITLE
Add recipe for s100py

### PR DESCRIPTION
**Pull request by erin-nagel**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/s100py/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    - pip
+  run:
+    - python >=3.8
+    - h5py
+    - numpy
+    - gdal >=3.2
+
+test:
+  requires:
+    - pytest
```
Comment (synapticarbors) [inline recipes/s100py/meta.yaml:31]: You shouldn't need to include `pytest`  unless you are actually running a test suite

File: recipes/s100py/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    - setuptools
+    - setuptools-scm
+    - pip
+  run:
+    - python >=3.8
+    - h5py
+    - numpy
+    - gdal >=3.2
+
+test:
```
Comment (synapticarbors) [inline recipes/s100py/meta.yaml:29]: You might want to include a `pip check`:
https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L61-L64

**Review by synapticarbors — COMMENTED** [review]
Just a few minor review questions/suggestions. I'll also need the listed co-maintainer to leave a comment or for you to confirm that you have gotten their ok offline.

**Comment by barry-gallagher** [comment]
Hi @synapticarbors , I'm the co-maintainer of s100py and agree with the work @erin-nagel has done to create the pull request and to maintain the conda-forge recipe in the future.

File: recipes/s100py/meta.yaml (hunk 1 above)
Comment (erin-nagel) [reply recipes/s100py/meta.yaml]: Thanks, we have removed pytest

File: recipes/s100py/meta.yaml (hunk 2 above)
Comment (synapticarbors) [reply recipes/s100py/meta.yaml:29]: Sorry for letting this fall off of my radar. It's probably helpful to add the `pip check`, but if you just want me to merge this, I'm fine with that too.

Comment (erin-nagel) [reply recipes/s100py/meta.yaml:29]: Thanks, let's go ahead with the merge.

**Pull request merged**